packages/analytics-js-integrations/src/integrations/MoEngage/browser.js

```diff
@@ -39,90 +56,167 @@ class MoEngage {
   }
 
   init() {
-    loadNativeSdk();
-    // setting the region if us then not needed.
-    if (this.region !== 'US') {
-      this.moeClient = window.moe({
-        app_id: this.apiId,
-        debug_logs: this.debug ? 1 : 0,
-        cluster: this.region === 'EU' ? 'eu' : 'in',
-      });
-    } else {
-      this.moeClient = window.moe({
-        app_id: this.apiId,
-        debug_logs: this.debug ? 1 : 0,
-      });
-    }
+    loadNativeSdk(this.calculateMoeDataCenter());
+    // Following MoEngage official convention: assign moe() result to global Moengage
+    window.Moengage = window.moe({
+      app_id: this.apiId,
+      debug_logs: this.debug ? 1 : 0,
+    });
+
     this.initialUserId = this.analytics.getUserId();
   }
 
   isLoaded() {
-    return !!window.moeBannerText;
+    // Check if MoEngage is properly initialized following their official pattern
+    return !!window.Moengage && typeof window.Moengage.track_event === 'function';
   }
 
   isReady() {
     return this.isLoaded();
   }
 
+  calculateMoeDataCenter() {
+    // Calculate the MoEngage data center based on the region
+    switch (this.region) {
+      case 'EU':
+        return 'dc_2';
+      case 'IN':
+        return 'dc_3';
+      default:
+        return 'dc_1'; // Default to US data center
+    }
+  }
+
   track(rudderElement) {
     // Check if the user id is same as previous session if not a new session will start
     if (!rudderElement.message) {
       logger.error('Payload not correct');
       return;
     }
+
     const { event, properties, userId } = rudderElement.message;
     if (userId && this.initialUserId !== userId) {
-      this.reset();
+      this.reset().then(() => {
+        // Continue after reset is complete
+        this.trackEvent(event, properties);
+      });
+      return;
     }
+    this.trackEvent(event, properties);
+  }
+
+  trackEvent(event, properties) {
     // track event : https://docs.moengage.com/docs/tracking-events
     if (!event) {
       logger.error('Event name is not present');
       return;
     }
     if (properties) {
-      this.moeClient.track_event(event, properties);
+      window.Moengage.track_event(event, properties);
     } else {
-      this.moeClient.track_event(event);
+      window.Moengage.track_event(event);
     }
   }
 
   reset() {
```

Comment:
can the promise be handled in the function itself instead of each caller to handle it explicitly ?